runtime/runtime-params-estimator/src/estimator_context.rs

```diff
@@ -156,14 +156,58 @@ impl<'c> EstimatorContext<'c> {
             &mut store_update,
             shard_uid,
             FlatStorageStatus::Ready(FlatStorageReadyStatus {
-                flat_head: BlockInfo::genesis(FLAT_STATE_HEAD, 0),
+                flat_head: BlockInfo::genesis(CryptoHash::hash_borsh(0usize), 0),
             }),
         );
         store_update.commit().expect("failed to set flat storage status");
         let flat_storage = FlatStorage::new(store, shard_uid);
+        self.generate_deltas(&flat_storage);
         flat_storage_manager.add_flat_storage_for_shard(shard_uid.shard_id(), flat_storage);
         flat_storage_manager
     }
+
+    /// Construct a chain of fake blocks with fake deltas for flat storage.
+    ///
+    /// Use `hash(height)` as the supposed block hash.
+    /// Keys are randomly generated, values are a constant that's not even stored.
+    /// 
+    /// The blocks aren't valid, nor are the values stored anywhere. They only
+    /// exist within `FlatStorage` and simulate the performance decrease
+    /// observed when the flat head lags behind.
+    fn generate_deltas(&self, flat_storage: &FlatStorage) {
+        // Assumption: One delta per non-final block, which is configurable.
+        // There could be forks but that's considered to e outside the normal
+        // operating conditions for this estimation.
+        let num_deltas = self.config.finality_lag;
+        // Number of keys changed is the same for all deltas and configurable.
+        let num_changes_per_delta = self.config.fs_keys_per_delta;
+        // This is the longest key we allow in storage.
+        let delta_key_len = 2000;
+        for idx in 0..num_deltas {
+            // We want different keys and to avoid all optimization potential.
+            // But the values are never read, so let's just use a dummy constant.
+            let random_data = iter::repeat_with(|| {
+                (
+                    crate::utils::random_vec(delta_key_len),
+                    Some(ValueRef::new(b"this is never stored or accessed, we only need it to blow up in-memory deltas")),
+                )
+            })
+            .take(num_changes_per_delta);
+            let height = 1 + idx as u64;
+            let block = BlockInfo {
+                hash: CryptoHash::hash_borsh(height),
+                height,
+                prev_hash: CryptoHash::hash_borsh(height - 1),
```

Comment:
nit: while it wasn't too hard for me to understand how this combination of `hash` and `prev_hash` generates a valid chain along with the fact that current flat head is at block `CryptoHash::hash_borsh(0)`, I still think it would be nice to avoid sorcery and keep it straightforward 😄 
one way is to add a parameter for flat head `fn generate_deltas(..., flat_head: CryptoHash)` and then use a mutable variable to store previous hash: 
```
let mut prev_hash = flat_head;
for idx in 0..num_deltas {
  ...
  let hash = CryptoHash::hash_borsh(height);
  let block = BlockInfo{ hash, height, prev_hash };
  prev_hash = hash;
}
```

Reply:
Hm, I see what you mean with sorcery :)

But together with the call in `fn trie()` it's a bit annoying to avoid. So far, we avoided state on `EstimatorContext` as much as possible, especially mutable state (other than lazy initialized caches with are only pseudo-mutable).

We also generate a bunch of accounts in a similar fashion, where we generate account ids for numbers 0..N and then implicitly assume in each `Testbed` instance that all those accounts exist. I think to keep the architecture of `EstimatorContext` / `Testbed` / `TransactionBuilder` aligned to its original design, we should do it in similar fashion here. 

Just for context, the account IDs are derived here:

https://github.com/near/nearcore/blob/c986e8a1b36b80ea4f20ad8e60b82017ab17a9f6/genesis-tools/genesis-populate/src/lib.rs#L42-L49

So for this PR, I now added a function 
```rust
/// Maps fake block heights to block hashes.
///
/// This is ued to generate and access fake deltas for flat storage.
fn fs_fake_block_height_to_hash(height: u64) -> CryptoHash {
    CryptoHash::hash_borsh(height)
}
```
to make it clearer how the mapping works, and I call it here and in the other location you pointed it out.

That keeps the estimator code consistent but I hope it also makes it look less like wizardry.